router/batchrouter/handle.go

```diff
@@ -556,6 +532,67 @@ func (brt *Handle) pingWarehouse(batchJobs *BatchedJobs, output UploadResult) (e
 	return
 }
 
+func (brt *Handle) generateSchemaMap(batchJobs *BatchedJobs) map[string]map[string]string {
+	// First group jobs by table name
+	jobsByTable := make(map[string][]*jobsdb.JobT)
+	metaDataByJob := make(map[int64]map[string]gjson.Result)
+	for _, job := range batchJobs.Jobs {
+		metadata := gjson.GetBytes(job.EventPayload, "metadata").Map()
+		tableName := metadata["table"].String()
+		if tableName == "" {
+			continue
+		}
+		jobsByTable[tableName] = append(jobsByTable[tableName], job)
+		metaDataByJob[job.JobID] = metadata
+	}
+
+	// Create channels for results
+	type tableResult struct {
+		tableName string
+		schema    map[string]string
+	}
+	resultChan := make(chan tableResult, len(jobsByTable))
+	g, _ := errgroup.WithContext(context.Background())
```

Comment:
Since we are not using context here in any way?
```suggestion
	g := errgroup.Group{}
```